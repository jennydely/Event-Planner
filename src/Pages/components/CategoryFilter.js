import React from 'react'
import SelectCategory from './common/SelectCategory'

export default function CategoryFilter({ onSelectFilter }) {
  const categories = [
    'category',
    'all',
    'metal',
    'medieval',
    'holiday',
    'other',
  ]

  return (
    <SelectCategory
      data-testid="my-select-component"
      id="category"
      placeholder="category"
      name={categories}
      options={categories}
      onSelectFilter={onSelectFilter}
    />
  )
}
