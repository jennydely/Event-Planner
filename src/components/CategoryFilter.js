import React from 'react'
import styled from 'styled-components/macro'
import SelectCategory from '../common/SelectCategory'

export default function CategoryFilter({ onSelectFilter }) {
  const categories = ['all', 'metal', 'medieval', 'holiday', 'other']

  return (
    <FilterContainer data-testid="my-select-component">
      <SelectCategory
        id="category"
        placeholder="all"
        name={categories}
        options={categories}
        onSelectFilter={onSelectFilter}
      />
    </FilterContainer>
  )
}

const FilterContainer = styled.div`
  margin: 5px;
`
