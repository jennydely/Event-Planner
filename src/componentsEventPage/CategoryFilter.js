import React from 'react'
import styled from 'styled-components/macro'
import SelectCategory from '../common/SelectCategory'


export default function CategoryFilter({ onSelectFilter }) {
 
  return (
    <FilterContainer>
      <SelectCategory id="category" name={["metal", "medieval", "sand", "other"]} 
        options={["metal", "medieval", "sand", "other"]}  onSelectFilter={onSelectFilter}/>
      <CategoryButton name="metal" onClick={() => onSelectFilter('metal')}>metal</CategoryButton>
      <CategoryButton name="medieval" onClick={() => onSelectFilter('medieval')}>medieval</CategoryButton>
      <CategoryButton name="sand" onClick={() => onSelectFilter('sand')}>sand</CategoryButton>
      <CategoryButton name="other" onClick={() => onSelectFilter('other')}>other</CategoryButton>
    </FilterContainer>
  )
}

const FilterContainer = styled.div`
margin 5px;
backgroundcolor: grey;
`

const CategoryButton = styled.button`
background: ${({ name }) => name === 'sand' ? 'rgba(248,149,17,0.46)' : (name === 'metal' ? 'rgba(49,42,42,0.75)' : (name === 'medieval' ? 'rgba(67,40,24,0.70)' : (name === 'other' ? 'rgba(153,88,42,0.70)' : 'rgb(96,99,104)')))};
` 