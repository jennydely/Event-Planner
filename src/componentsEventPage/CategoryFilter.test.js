import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import CategoryFilter from './CategoryFilter'
import 'jest-styled-components'

describe('CategoryFilter', () => {
    const onSelectFilter = 'medieval'

    it('displays all buttons to filter', () => {
        const { getByText } = render(<CategoryFilter onSelectFilter={onSelectFilter} />)
        expect(getByText('all')).toBeInTheDocument()
        expect(getByText('medieval')).toBeInTheDocument()
        expect(getByText('metal')).toBeInTheDocument()
        expect(getByText('sand')).toBeInTheDocument()
        expect(getByText('other')).toBeInTheDocument()
    })
    it('renders correctly', () => {
        const tree = renderer.create(<CategoryFilter onSelectFilter={onSelectFilter} />)
        expect(tree).toMatchSnapshot()
    })
})