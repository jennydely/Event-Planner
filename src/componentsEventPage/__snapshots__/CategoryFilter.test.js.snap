// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter renders correctly 1`] = `
.c1 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: 1px solid var(--blue-50);
  margin-top: 0;
  padding: 4px;
  font-size: 100%;
  color: black;
  background: rgb(96,99,104);
}

.c2 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: 1px solid var(--blue-50);
  margin-top: 0;
  padding: 4px;
  font-size: 100%;
  color: black;
  background: rgb(96,99,104);
}

.c3 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: 1px solid var(--blue-50);
  margin-top: 0;
  padding: 4px;
  font-size: 100%;
  color: black;
  background: rgba(49,42,42,0.75);
}

.c4 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: 1px solid var(--blue-50);
  margin-top: 0;
  padding: 4px;
  font-size: 100%;
  color: black;
  background: rgba(67,40,24,0.70);
}

.c5 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: 1px solid var(--blue-50);
  margin-top: 0;
  padding: 4px;
  font-size: 100%;
  color: black;
  background: rgba(248,149,17,0.46);
}

.c6 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: 1px solid var(--blue-50);
  margin-top: 0;
  padding: 4px;
  font-size: 100%;
  color: black;
  background: rgba(153,88,42,0.70);
}

.c0 {
  margin: 5px;
  backgroundcolor: grey;
}

<div
  className="c0"
>
  <select
    className="c1"
    id="category"
    name={
      Array [
        "all",
        "metal",
        "medieval",
        "sand",
        "other",
      ]
    }
  >
    <option
      className="c2"
      onClick={[Function]}
      value="all"
    >
      all
    </option>
    <option
      className="c3"
      onClick={[Function]}
      value="metal"
    >
      metal
    </option>
    <option
      className="c4"
      onClick={[Function]}
      value="medieval"
    >
      medieval
    </option>
    <option
      className="c5"
      onClick={[Function]}
      value="sand"
    >
      sand
    </option>
    <option
      className="c6"
      onClick={[Function]}
      value="other"
    >
      other
    </option>
  </select>
</div>
`;
